File: app/src/main/java/com/bth/running/util/RunUtils.java
package com.bth.running.util;

import org.joda.time.Period;
import org.joda.time.PeriodType;
import org.joda.time.format.PeriodFormatter;
import org.joda.time.format.PeriodFormatterBuilder;


/**
 * @author Martin Macheiner
 *         Date: 10.09.2017.
 */

public class RunUtils {

    private static final double METS_5_30_PACE = 12.8;

    public static String calculatePace(long timeInMs, double distance) {

        if (distance <= 0) {
            return "-:--";
        }
        long kmMillis = (long) (timeInMs / distance);
        return formatPaceMillisToString(kmMillis);
    }

    public static int calculateCaloriesBurned(double runningTimeInMillis, double weightOfRunner) {
        //double burned = distance * weightOfRunner * 1.036;
        double runningTimeInHours = runningTimeInMillis / 3600000d;
        double burned = METS_5_30_PACE * weightOfRunner * runningTimeInHours;
        return (int) Math.floor(burned);
    }

    private static String formatPaceMillisToString(long kmMillis) {

        PeriodFormatter formatter = new PeriodFormatterBuilder()
                .minimumPrintedDigits(2)
                .appendMinutes()
                .appendSeparator(":")
                .appendSeconds()
                .toFormatter();

        PeriodType minutesSeconds = PeriodType.time()
                .withMillisRemoved()
                .withHoursRemoved();

        Period kmPeriod = new Period(kmMillis, minutesSeconds);
        return formatter.print(kmPeriod);
    }

}

File: app/src/main/java/at/shockbytes/corey/running/util/RunUtils.kt
package at.shockbytes.corey.running.util

import org.joda.time.Period
import org.joda.time.PeriodType
import org.joda.time.format.PeriodFormatter
import org.joda.time.format.PeriodFormatterBuilder
import java.text.SimpleDateFormat
import java.util.*


/**
 * @author Martin Macheiner
 * Date: 10.09.2017.
 */

object RunUtils {

    private const val METS_5_30_PACE = 12.8

    val periodFormatter: PeriodFormatter
        get() = PeriodFormatterBuilder()
                .minimumPrintedDigits(2)
                .printZeroAlways()
                .appendHours()
                .appendSeparator(":")
                .appendMinutes()
                .appendSeparator(":")
                .appendSeconds()
                .toFormatter()

    fun getDayOfWeekAbbr(date: Long): String {
        val locale = Locale.US
        val format2 = SimpleDateFormat("EE", locale)
        return format2.format(Date(date)).toUpperCase(locale)
    }

    fun calculatePace(timeInMs: Long, distance: Double): String {

        if (distance <= 0) {
            return "-:--"
        }
        val kmMillis = (timeInMs / distance).toLong()
        return formatPaceMillisToString(kmMillis)
    }

    fun calculateCaloriesBurned(runningTimeInMillis: Double, weightOfRunner: Double): Int {
        //double burned = distance * weightOfRunner * 1.036;
        val runningTimeInHours = runningTimeInMillis / 3600000.0
        val burned = METS_5_30_PACE * weightOfRunner * runningTimeInHours
        return Math.floor(burned).toInt()
    }

    private fun formatPaceMillisToString(kmMillis: Long): String {

        val formatter = PeriodFormatterBuilder()
                .minimumPrintedDigits(2)
                .appendMinutes()
                .appendSeparator(":")
                .appendSeconds()
                .toFormatter()

        val minutesSeconds = PeriodType.time()
                .withMillisRemoved()
                .withHoursRemoved()

        val kmPeriod = Period(kmMillis, minutesSeconds)
        return formatter.print(kmPeriod)
    }

}
